### PR TITLE
Fixing pushDeploymentDataToEvidenceStore logic

### PR DESCRIPTION
**Pull request by shigupt202**
Fixes https://github.com/microsoft/azure-pipelines-tasks/issues/12798 and https://github.com/microsoft/azure-pipelines-tasks/issues/11931

1. The new logic extracts release name from the helm output. It would work whether --debug flag is used or not. The release name is then used to fetch the manifests in this release using `helm get manifest RELEASE_NAME` command.

2. Added e2e tests for HelmDeployV0

### CODE REVIEW

File: Tasks/HelmDeployV0/src/utils.ts [hunk 1]
```diff
@@ -41,14 +46,43 @@ export function resolvePath(path : string): string {
...
+    return '';
+}
+
+export function getManifestsFromRelease(helmCli: helmcli, releaseName: string): any {
+    let manifests = [];
+    if (releaseName.length == 0)
+        return manifests;
+
+    helmCli.resetArguments();
+    helmCli.setCommand('get');
```
[the remark this answers is not in the document]
Comment (shigupt202) [reply Tasks/HelmDeployV0/src/utils.ts:68]: Yes

File: Tasks/HelmDeployV0/src/utils.ts [hunk 2]
```diff
@@ -41,14 +46,43 @@ export function resolvePath(path : string): string {
...
+    helmCli.addArgument('manifest');
+    helmCli.addArgument(releaseName);
+
+    const execResult = helmCli.execHelmCommand();
+    const files = execResult.stdout.split("---");
```
[the remark this answers is not in the document]
Comment (shigupt202) [reply Tasks/HelmDeployV0/src/utils.ts]: Done

File: Tasks/HelmDeployV0/src/helm.ts [hunk 3]
```diff
@@ -165,7 +165,8 @@ function runHelm(helmCli: helmcli, command: string, kubectlCli: kubernetescli, f
     else if ((command === "install" || command === "upgrade")) {
```
Comment (ds-ms) [inline Tasks/HelmDeployV0/src/helm.ts:165]: Why are there double brackets?

File: Tasks/HelmDeployV0/Tests/L0.ts [hunk 4]
```diff
@@ -1,16 +1,194 @@
...
+        process.env[shared.TestEnvVars.connectionType] = shared.ConnectionTypes.KubernetesServiceConnection;
+        process.env[shared.TestEnvVars.command] = shared.Commands.install;
+        process.env[shared.TestEnvVars.chartType] = shared.ChartTypes.Name;
+        process.env[shared.TestEnvVars.chartName] = shared.testChartName;
+        process.env[shared.TestEnvVars.releaseName] = shared.testReleaseName;
+        process.env[shared.TestEnvVars.failOnStderr] = 'ture';
+        process.env[shared.isHelmV3] = 'true';
+
+        tr.run();
+        assert(tr.stdout.indexOf("v3") != -1, 'Helm version 3 installed');
```
Comment (ds-ms) [inline Tasks/HelmDeployV0/Tests/L0.ts:48]: inconsistent quotes across the file

File: Tasks/HelmDeployV0/Tests/L0.ts [hunk 5]
```diff
@@ -1,16 +1,194 @@
...
+        process.env[shared.TestEnvVars.connectionType] = shared.ConnectionTypes.KubernetesServiceConnection;
+        process.env[shared.TestEnvVars.command] = shared.Commands.install;
+        process.env[shared.TestEnvVars.chartType] = shared.ChartTypes.Name;
+        process.env[shared.TestEnvVars.chartName] = shared.testChartName;
+        process.env[shared.TestEnvVars.releaseName] = shared.testReleaseName;
+        process.env[shared.TestEnvVars.failOnStderr] = 'false';
+        process.env[shared.TestEnvVars.namespace] = shared.testNamespace;
+
+        tr.run();
+        assert(tr.stdout.indexOf("v2") != -1, 'Helm version 2 installed');
```
Comment (ds-ms) [inline Tasks/HelmDeployV0/Tests/L0.ts:68]: NIT: Shouldn't the message say `Helm v2 should have been installed`?

File: Tasks/HelmDeployV0/Tests/L0.ts [hunk 6]
```diff
@@ -1,16 +1,194 @@
...
+        assert(tr.stdout.indexOf('STATUS: deployed') != -1, `Release created with NAME: ${shared.testReleaseName}`);
+        assert(tr.stdout.indexOf('# Source: testChartName/templates/serviceaccount.yaml') != -1, `Manifests extracted from release ${shared.testReleaseName}`);
+        assert(tr.stdout.indexOf('DeploymentDetailsApiResponse: {"mockKey":"mockValue"}') != -1, 'Web response for pushing metadata to evidence store');
```
Comment (ds-ms) [inline Tasks/HelmDeployV0/Tests/L0.ts:71]: The message you're writing would be printed if the assert fails. So ideally you should frame those sentences in a way to show that the desired state wasn't acheived.

File: Tasks/HelmDeployV0/Tests/L0.ts [hunk 7]
```diff
@@ -1,16 +1,194 @@
...
+    it('Helm init should fail (version 3)', function(done: MochaDone) {        
+        const tp = path.join(__dirname, 'TestSetup.js');
+        const tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+        process.env[shared.TestEnvVars.connectionType] = shared.ConnectionTypes.KubernetesServiceConnection;
+        process.env[shared.TestEnvVars.command] = shared.Commands.init;
+        process.env[shared.TestEnvVars.failOnStderr] = 'false';
+        process.env[shared.isHelmV3] = 'true';
+
+        tr.run();
+        assert(tr.failed, 'task should have failed');
```
Comment (ds-ms) [inline Tasks/HelmDeployV0/Tests/L0.ts:174]: Add an assertion for the expected message

File: Tasks/HelmDeployV0/Tests/TestSetup.ts [hunk 8]
```diff
@@ -0,0 +1,314 @@
...
+tr.setInput('save', process.env[shared.TestEnvVars.save] || '');
+tr.setInput('install', process.env[shared.TestEnvVars.install] || '');
+tr.setInput('recreate', process.env[shared.TestEnvVars.recreate] || '');
+tr.setInput('resetValues', process.env[shared.TestEnvVars.resetValues] || '');
+tr.setInput('force', process.env[shared.TestEnvVars.force] || '');
+tr.setInput('waitForExecution', process.env[shared.TestEnvVars.waitForExecution] || '');
+tr.setInput('arguments', process.env[shared.TestEnvVars.arguments] || '');
+tr.setInput('failOnStderr', process.env[shared.TestEnvVars.failOnStderr] || 'true');
+tr.setInput('chartNameForACR', process.env[shared.TestEnvVars.chartNameForACR] || '');
+tr.setInput('chartPathForACR', process.env[shared.TestEnvVars.chartPathForACR] || '');
```
Comment (ds-ms) [inline Tasks/HelmDeployV0/Tests/TestSetup.ts:56]: These are a lot of inputs, we should discuss a strategy for reducing these.

File: Tasks/HelmDeployV0/Tests/L0.ts (hunk 4 above)
Comment (shigupt202) [reply Tasks/HelmDeployV0/Tests/L0.ts:48]: Will update to double quotes everywhere. However, some strings need to use `  as they are using variables.

File: Tasks/HelmDeployV0/Tests/L0.ts (hunk 5 above)
Comment (shigupt202) [reply Tasks/HelmDeployV0/Tests/L0.ts:68]: Yes. I'll update.

File: Tasks/HelmDeployV0/Tests/L0.ts (hunk 6 above)
Comment (shigupt202) [reply Tasks/HelmDeployV0/Tests/L0.ts:71]: Got it. Will change them.

File: Tasks/HelmDeployV0/Tests/L0.ts (hunk 7 above)
Comment (shigupt202) [reply Tasks/HelmDeployV0/Tests/L0.ts:174]: `helm init` is not an expected command in Helm 3. So, there's no error message. It just prints the available commands. Should I write an assert for that?

File: Tasks/HelmDeployV0/src/helm.ts (hunk 3 above)
Comment (shigupt202) [reply Tasks/HelmDeployV0/src/helm.ts:165]: They shouldn't be there. I'll remove them.

**Pull request merged**